Messages by bots do not have a user

**Describe the bug**
When searching, messages written by apps do not have a user set and fail by throwing:

> Cannot build Message, some of required attributes are not set [user]

**To Reproduce**
Steps to reproduce the behavior:
1. Search for messages in some channel written by an app:
```
searchMessages(SearchMessagesParams.builder()
                .setQuery("in:#some_channel written by app")
                .build())
                .join()
                .unwrapOrElseThrow()
                .messages
                .matches
                .firstOrNull()
```

2. Jackson tries to parse found message:
```
{
      "iid" : "...",
      "team" : "...",
      "channel" : {
        "id" : "...",
        "is_channel" : true,
        "is_group" : false,
        "is_im" : false,
        "name" : "some_channel",
        "is_shared" : false,
        "is_org_shared" : false,
        "is_ext_shared" : false,
        "is_private" : false,
        "is_mpim" : false,
        "pending_shared" : [ ],
        "is_pending_ext_shared" : false
      },
      "type" : "message",
      "user" : null,
      "username" : "app-name",
      "ts" : "1572520802.000200",
      "text" : "written by an app",
      "permalink" : "..."
    }
```
3. Exception is thrown:
```
Caused by: com.hubspot.immutables.validation.InvalidImmutableStateException: Cannot build Message, some of required attributes are not set [user]
	at com.hubspot.slack.client.models.Message$Builder.checkRequiredAttributes(Message.java:1298)
	at com.hubspot.slack.client.models.Message$Builder.build(Message.java:1251)
	at com.hubspot.slack.client.models.Message.fromJson(Message.java:921)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.fasterxml.jackson.databind.introspect.AnnotatedMethod.call1(AnnotatedMethod.java:114)
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator._createUsingDelegate(StdValueInstantiator.java:516)
	... 35 more
Exception in thread "main" java.lang.IllegalStateException: SlackError{type=JSON_PARSING_FAILED, error=_json_parsing_failed}
	at com.hubspot.algebra.Result.lambda$unwrapOrElseThrow$1(Result.java:68)
	at com.hubspot.algebra.Result.lambda$unwrapOrElseThrow$0(Result.java:64)
	at java.util.Optional.orElseThrow(Optional.java:290)
	at com.hubspot.algebra.Result.unwrapOrElseThrow(Result.java:60)
	at com.hubspot.algebra.Result.unwrapOrElseThrow(Result.java:64)
	at com.hubspot.algebra.Result.unwrapOrElseThrow(Result.java:68)
```

**Expected behavior**
Does not crash. User shouldn't be required?

**Additional context**
It's interesting that the message's `username` is present but not the `user`. Are they similar?

Message was previously written by the same script using a slack token for the app:

```
val config = SlackClientRuntimeConfig.builder()
    .setTokenSupplier { slackToken }
    .build()
val slackClient = SlackClientFactory.defaultFactory().build(config)

postMessage(ChatPostMessageParams.builder()
    .setText(message)
    .setChannelId(slackChannel)
    .build())
    .join()
    .unwrapOrElseThrow()
```

--- a/slack-base/src/main/java/com/hubspot/slack/client/models/MessageIF.java
+++ b/slack-base/src/main/java/com/hubspot/slack/client/models/MessageIF.java
@@ -16,7 +16,7 @@ public interface MessageIF {
   String getTeam();
   SlackChannel getChannel();
 
-  String getUser();
+  Optional<String> getUser();
   String getUsername();
 
   @JsonProperty("ts")

